core/src/main/java/bio/terra/pearl/core/service/notification/email/EnrolleeEmailService.java

```diff
@@ -154,6 +155,11 @@ public boolean shouldSendEmail(Trigger config,
                     ruleData.getEnrollee().getShortcode(), config.getId(), config.getPortalEnvironmentId());
             return false;
         }
+        if (ruleData.getProfile() != null && ruleData.getProfile().isDoNotEmailSolicit() && config.getTaskType().equals(TaskType.OUTREACH)) {
```

Comment:
can we factor out the `ruleData.getProfile() != null` check?  Also, rename `ruleData` to `enrolleeContext` -- we're consolidating rule and search contexts, so that will be clearer going forward.
```
Profile profile = enrolleeContext.getProfile()
if (profile == null) {
  return false;  // no address to send email to
}
```